Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.h

```diff
@@ -214,6 +214,16 @@ namespace AzToolsFramework
 
             AZStd::string PrefabDomValueToString(const PrefabDomValue& prefabDomValue);
 
+            //! Adds a nested instance to the the prefab DOM if it doesn't already exist and optionally update its contents.
+            //! @param prefabDom The prefab DOM to udpate.
+            //! @param nestedInstanceAlias The alias of the nested instance to be added or updated.
+            //! @param nestedInstanceDom An optional value to assign to the nested instance in the prefab DOM.
+            //! @return bool on whether the prefab DOM was updated.
+            bool AddOrUpdateNestedInstance(
```

Comment:
Suggestion: Could this method be split into two "Add" and "Update" methods. 
Reasoning : Right now debugging this method would involve watching out for both Add and Update cases. Making those operations distinct and separated out would lead to less cognitive load especially if you are debugging a complex prefab with multiple nested instances , some of which are being updated and others being added.

Reply:
That would require the caller knowing whether the nested instance already exists in the DOM or not... I'll give it some thought.